draft-ietf-tls-esni.md

```diff
@@ -274,25 +280,27 @@ constrain server name lengths. Names may exceed this length if, e.g.,
 the server uses wildcard names or added new names to the anonymity set.
 
 public_name
-: The non-empty name of the client-facing server, i.e., the entity trusted to
-update the ECH configuration. This is used to correct misconfigured clients, as
-described in {{handle-server-response}}. This value MUST NOT begin or end with
-an ASCII dot and MUST be parsable as a dot-separated sequence of LDH labels, as
-defined in {{!RFC5890, Section 2.3.1}}. Clients MUST ignore any `ECHConfig`
-structure whose `public_name` does not meet these criteria. Note that these
-criteria are incomplete; they incidentally rule out textual representations of
-IPv6 addresses (see {{!RFC3986, Section 3.2.2}}), but do not exclude IPv4
-addresses in standard dotted-decimal or other non-standard notations such as
-octal and hexadecimal (see {{RFC3986, Section 7.4}}). If `public_name` contains
-a literal IPv4 or IPv6 address, the client SHOULD ignore the `ECHConfig` to
-avoid sending a non-compliant "server_name" extension on the ClientHelloOuter
-(see {{!RFC6066, Section 3}}).
+: The DNS name of the client-facing server, i.e., the entity trusted
+to update the ECH configuration. This is used to correct misconfigured clients,
+as described in {{handle-server-response}}. This value MUST NOT begin or end
+with an ASCII dot and MUST be parsable as a dot-separated sequence of LDH
+labels, as defined in {{!RFC5890, Section 2.3.1}}. Clients MUST ignore any
+`ECHConfig` structure whose public_name does not meet these criteria. Note
+that these criteria allow IPv4 addresses in dotted-decimal or other notations,
+e.g. {{?RFC3986, Section 7.4}} and {{WHATWG-IPV4}}. Clients SHOULD ignore the
+`ECHConfig` if it contains an encoded IPv4 address. (To determine if a
+public_name value is an IPv4 address, clients can invoke the IPv4 parser
+algorithm in {{WHATWG-IPV4}}. It returns a value when the input is an IPv4
+address.) See {{auth-public-name}} for how the client interprets and validates
+the public_name.
```

Comment:
Try splitting this up a little.  You can start multiple sections with ": " to make separate paragraphs (I think).

Reply:
Oh, nice. I didn't know that was possible. I broke up the description into logical units.